src/helper/ecdsa.rs

```diff
@@ -0,0 +1,223 @@
+        let key = self.into();
+        let p = Mpi::from_raw(self.to_sec1_bytes().to_vec());
+        EcdsaPublicParams::P256 { key, p }
+    }
+}
+
+/// [`signature::Signer`] backed signer for PGP.
+pub struct EcdsaSigner<T, C> {
+    inner: T,
+    public_key: PublicKey,
+    _signature: PhantomData<C>,
+}
+
+impl<C, T> EcdsaSigner<T, C>
+where
+    T: Keypair,
+    T::VerifyingKey: PgpPublicKey,
+{
+    /// Create a new signer with a given public key
+    pub fn new(inner: T, created_at: DateTime<Utc>) -> Result<Self> {
+        let public_key = PublicKey::new(
+            Version::New,
+            KeyVersion::V4,
+            <T as Keypair>::VerifyingKey::PGP_ALGORITHM,
+            created_at,
+            None,
+            inner.verifying_key().pgp_parameters(),
+        )?;
+
+        Ok(Self {
+            inner,
+            public_key,
+            _signature: PhantomData,
+        })
+    }
+}
+
+impl<C, T> Keypair for EcdsaSigner<T, C>
+where
+    T: Keypair,
+{
+    type VerifyingKey = T::VerifyingKey;
+
+    fn verifying_key(&self) -> Self::VerifyingKey {
+        self.inner.verifying_key()
+    }
+}
+
+impl<C, T> EcdsaSigner<T, C>
+where
+    C: PrimeCurve + DigestPrimitive,
+    SignatureSize<C>: ArrayLength<u8>,
+    T: PrehashSigner<ecdsa::Signature<C>>,
+    C::Digest: PgpHash,
+{
+    fn sign_prehash(&self, hash: HashAlgorithm, prehash: &[u8]) -> Result<Vec<Vec<u8>>> {
+        if C::Digest::HASH_ALGORITHM != hash {
+            bail!(
+                "Signer only support {expected:?}, found {found:?}",
+                expected = C::Digest::HASH_ALGORITHM,
+                found = hash
+            );
+        }
+        // TODO: check digest size
+        let signature = self.inner.sign_prehash(prehash)?;
+        let (r, s) = signature.split_bytes();
+        Ok(vec![r.to_vec(), s.to_vec()])
+    }
+}
+
+impl<C, T> fmt::Debug for EcdsaSigner<T, C> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("EcdsaSigner").finish()
+    }
+}
+
+impl<C, T> SecretKeyTrait for EcdsaSigner<T, C>
+where
+    C: PrimeCurve + DigestPrimitive,
+    SignatureSize<C>: ArrayLength<u8>,
+    T: PrehashSigner<ecdsa::Signature<C>>,
+    C::Digest: PgpHash,
+{
+    type PublicKey = PublicKey;
+    type Unlocked = Self;
+
+    fn unlock<F, G, Tr>(&self, _pw: F, work: G) -> Result<Tr>
+    where
+        F: FnOnce() -> String,
+        G: FnOnce(&Self::Unlocked) -> Result<Tr>,
+    {
+        work(self)
+    }
+
+    fn create_signature<F>(
+        &self,
+        _key_pw: F,
+        hash: HashAlgorithm,
+        prehashed_data: &[u8],
+    ) -> Result<SignatureBytes>
+    where
+        F: FnOnce() -> String,
+    {
+        let sig = self.sign_prehash(hash, prehashed_data)?;
+
+        // MPI format:
+        // strip leading zeros, to match parse results from MPIs
+        let mpis = sig
+            .iter()
+            .map(|v| Mpi::from_slice(&v[..]))
+            .collect::<Vec<_>>();
```

Comment:
I wonder if you need the explicit type here since `SignatureBytes::Mpis` takes a `Vec`:

```suggestion
            .collect();
```